statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,254 @@
+import statsmodels.tsa.base.tsa_model as tsbase
+import numpy as np
+import scipy
+
+# VECM class: for known or unknown VECM
+
+class VECM(tsbase.TimeSeriesModel):
+    r"""
+    Fit VECM process and do lag order selection
+
+    .. math:: y_t = A_1 y_{t-1} + \ldots + A_p y_{t-p} + u_t
+
+    Parameters
+    ----------
+    endog : array-like
+        2-d endogenous response variable. The independent variable.
+    dates : array-like
+        must match number of rows of endog
+
+    References
+    ----------
+    Lutkepohl (2005) New Introduction to Multiple Time Series Analysis
```

Comment:
Use RST markup for a bibliography entry.  Look in SM for examples.  Start with [1]